src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/Profile_PaymentMethodsScreen.tsx

```diff
@@ -235,6 +250,61 @@ const GenericError = () => {
   );
 };
 
+export const useOnRecurringPaymentReceived = ({
+  recurringPaymentId,
+  callback,
+}: {
+  recurringPaymentId: number | undefined;
+  callback: () => void;
+}) => {
+  const {userId} = useAuthContext();
+  const [recurringPayments, setRecurringPayments] = useState<
+    RecurringPayment[]
+  >([]);
+
+  const mapRecurringPayment = (
+    recurringPayment: FirebaseFirestoreTypes.DocumentData,
+  ): RecurringPayment => {
+    return {
+      id: recurringPayment.id,
+      expires_at: recurringPayment.expires_at,
+      masked_pan: recurringPayment.masked_pan,
+      payment_type: recurringPayment.payment_type,
+    };
+  };
+
+  const recurringPaymentsUnsub = firestore()
+    .collection('customers')
+    .doc(userId)
+    .collection('recurringPayments')
+    .orderBy('created', 'desc')
+    .onSnapshot(
+      (snapshot) => {
+        const recurringPayments =
+          snapshot.docs.map<RecurringPayment>(mapRecurringPayment);
+        setRecurringPayments(recurringPayments);
+      },
+      (err) => {
+        Bugsnag.notify(err, function (event) {
+          event.addMetadata('payment', {userId});
+        });
+      },
+    );
+
+  useEffect(() => {
+    if (
+      recurringPayments.some(
+        (rp) =>
+          rp.id === recurringPaymentId && recurringPaymentId !== undefined,
+      )
+    ) {
+      callback();
+    }
+
+    return recurringPaymentsUnsub;
+  }, [callback, recurringPaymentId, recurringPayments, recurringPaymentsUnsub]);
```

Comment:
Do we need this `useEffect` and the accompanying `useState` for the recurring payments? Or could we just invoke `callback` directly in the `onSnapshot` handler?

Reply:
Right! I made a fix to handle that better.